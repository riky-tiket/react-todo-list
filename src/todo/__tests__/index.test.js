import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Todo from '../../todo/index';
import Renderer from 'react-test-renderer';
import sinon from 'sinon';

Enzyme.configure({ adapter: new Adapter() });

describe('Todo Test UI', () => {
  let component = null;

  beforeEach(() => {
    component = (
      <Todo />
    );
  });

  it('Should render without error', () => {
    const tree = Renderer.create(component);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should have main-container class', () => {
    const wrapper = shallow(component);

    expect(wrapper.find('div .main-container').hasClass('main-container')).toBe(true);
  });

  it('Should have main-title class', () => {
    const wrapper = shallow(component);

    expect(wrapper.find('div .main-container .main-title').hasClass('main-title')).toBe(true);
  });

  it('Should have TODO title in main-title class', () => {
    const wrapper = shallow(component);
    const mainTitle = wrapper.find('div .main-container .main-title');

    expect(mainTitle.text()).toContain('TODO');
  })

  it('Should have new-todo class', () => {
    const wrapper = shallow(component);

    expect(wrapper.find('div .main-container .new-todo').hasClass('new-todo')).toBe(true);
  })

  it('Should have Add New Todo text in new-todo class', () => {
    const wrapper = shallow(component);
    const text = wrapper.find('div .main-container .new-todo');

    expect(text.text()).toContain('Add New Todo');
  })

  it('Should have clear-todo class', () => {
    const wrapper = shallow(component);

    expect(wrapper.find('div .main-container .clear-todo').hasClass('clear-todo')).toBe(true);
  })

  it('Should have Clear Todo List text in clear-todo class', () => {
    const wrapper = shallow(component);
    const text = wrapper.find('div .main-container .clear-todo');

    expect(text.text()).toContain('Clear Todo List');
  })
})

// TODO
// ngetes fungsi click