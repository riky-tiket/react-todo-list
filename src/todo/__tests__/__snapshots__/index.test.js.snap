// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Test UI Should render without error 1`] = `
<div>
  <div
    className="main-container"
  >
    <div
      className="main-title"
    >
      TODO
    </div>
    <div
      className="new-todo"
      onClick={[Function]}
    >
      Add New Todo
    </div>
    <div
      className="clear-todo"
      onClick={[Function]}
    >
      Clear Todo List
    </div>
    <div>
      <div
        className="todo-list"
        onClick={[Function]}
      >
        <span>
          1
           .
        </span>
        memasak
      </div>
      <div
        className="remove-todo"
        onClick={[Function]}
      >
        Hapus
      </div>
      <div
        className="pull-right"
        onClick={[Function]}
      >
        <button>
          +
        </button>
      </div>
      <div
        className="todo-list sub-todo"
        onClick={[Function]}
      >
        - 
        masak air
      </div>
      <div
        className="todo-list sub-todo"
        onClick={[Function]}
      >
        - 
        masak nasi
      </div>
    </div>
    <div>
      <div
        className="todo-list"
        onClick={[Function]}
      >
        <span>
          2
           .
        </span>
        belanja
      </div>
      <div
        className="remove-todo"
        onClick={[Function]}
      >
        Hapus
      </div>
      <div
        className="pull-right"
        onClick={[Function]}
      >
        <button>
          +
        </button>
      </div>
    </div>
    <div>
      <div
        className="todo-list"
        onClick={[Function]}
      >
        <span>
          3
           .
        </span>
        ngepel
      </div>
      <div
        className="remove-todo"
        onClick={[Function]}
      >
        Hapus
      </div>
      <div
        className="pull-right"
        onClick={[Function]}
      >
        <button>
          +
        </button>
      </div>
    </div>
    <div>
      <div
        className="todo-list"
        onClick={[Function]}
      >
        <span>
          4
           .
        </span>
        beberes rumah
      </div>
      <div
        className="remove-todo"
        onClick={[Function]}
      >
        Hapus
      </div>
      <div
        className="pull-right"
        onClick={[Function]}
      >
        <button>
          +
        </button>
      </div>
      <div
        className="todo-list sub-todo"
        onClick={[Function]}
      >
        - 
        kamar mandi
      </div>
      <div
        className="todo-list sub-todo"
        onClick={[Function]}
      >
        - 
        ruang tamu
      </div>
    </div>
  </div>
</div>
`;
